File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.java
class Solution {
    public int maxFreqSum(String s) {
        HashMap<Character,Integer> vowel = new HashMap<>();
        HashMap<Character,Integer> consonant = new HashMap<>();

        for(char c : s.toCharArray()){
            if(c == 'a' || c == 'e'||c == 'i' || c == 'o'||c == 'u'){
                vowel.put(c,vowel.getOrDefault(c,0)+1);
            }else{
                consonant.put(c,consonant.getOrDefault(c,0)+1);
            }
        }

        int a = 0 , b = 0;
        for(int i : vowel.values()){
            a = Math.max(i,a);
        }
        for(int i : consonant.values()){
            b = Math.max(i,b);
        }

        return a+b;
    }
}

File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.cpp
class Solution {
public:
    int maxFreqSum(string s) {
        unordered_map<char, int> vowels;
        unordered_map<char, int> c;

        for (auto x : s) {
            if (x == 'a' || x == 'e' || x == 'i' || x == 'o' || x == 'u') {
                vowels[x]++;
            } else {
                c[x]++;
            }
        }

        int v = 0;
        for (auto x : vowels) {
            if (x.second > v) {
                v = x.second;
            }
        }
        int cc = 0;

        for (auto x : c) {
            if (x.second > cc) {
                cc = x.second;
            }
        }

        return cc + v;
    }
};